pkg/registry/manifests.go

```diff
@@ -64,23 +68,31 @@ func GetAllImages(embeddedContainerImages []image.ContainerImage, manifestURLs [
 	return allImages, nil
 }
 
-func readManifest(manifestPath string) (any, error) {
-	manifestData, err := os.ReadFile(manifestPath)
+func readManifest(manifestPath string) ([]any, error) {
+	manifestFile, err := os.Open(manifestPath)
 	if err != nil {
-		return nil, fmt.Errorf("error reading manifest: %w", err)
+		return nil, fmt.Errorf("error opening manifest: %w", err)
 	}
 
-	if len(manifestData) == 0 {
-		return nil, fmt.Errorf("invalid manifest")
+	var manifests []any
+	decoder := yaml.NewDecoder(manifestFile)
+	for {
+		var manifest any
```

Comment:
I love that we're able to parse each YAML object one by one. Please consider using this structure in order to make the parsing step a little less "generic":
```suggestion
	var manifests []map[string]any
	decoder := yaml.NewDecoder(manifestFile)
	for {
		var manifest map[string]any
```

The output of the `readManifest` function will then be of type `[]map[string]any` and we should only need to make sure that `storeManifestImageNames` accepts it directly and works over it:
```
func storeManifestImageNames(data []map[string]any...
```

```
for _, d := range data {
	findImages(d)
}
```

Reply:
Done, thanks 👍